experiments/DaggerInstanceGLA/src/DaggerInstanceGLA.jl

```diff
@@ -0,0 +1,78 @@
+module DaggerInstanceGLA
+
+import Base: +
+using AutoHashEquals
+
+export DagDom, LinearMap, MatrixThunk, matrixToThunk,
+       dom, codom, adjoint, +, compose, id, oplus, mzero,
+       braid, mcopy, delete, plus, zero, scalar, antipode,
+       ⊗
+using Dagger
+using LinearAlgebra
+using Catlab.LinearAlgebra.GraphicalLinearAlgebra
+using Catlab, Catlab.Doctrines
+import ...Doctrines:
+  Ob, Hom, dom, codom, compose, ⋅, ∘, id, oplus, ⊕, mzero, braid,
+  dagger, dunit, dcounit, mcopy, Δ, delete, ◊, mmerge, ∇, create, □,
+  plus, zero, coplus, cozero, meet, top, join, bottom
+using LinearMaps
+import LinearMaps: adjoint
+const LMs = LinearMaps
+
+
+@auto_hash_equals struct DagDom
+  N::Int
+end
+
+# This structure was created to keep track of dom and codom information.
+# This information can be updated efficiently, and keeping it here keeps
+# LinearFunctions from having to think the thunk each time the dom or codom
+# is queried
+
+struct MatrixThunk
+  thunk::Thunk
+  dom::Int
+  codom::Int
+end
+
+matrixToThunk(A::LinearMap) = begin
```

Comment:
This function should be called `Thunk(A::LinearMap)` and you add `import Dagger: Thunk` to the top of the file